### PR TITLE
Fix AWS Console apps logins not being provided in all places

### LINKED ISSUE #45086: `TestListUnifiedResources_WithLogins ` flakiness
#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/10147423334/job/28057746121
- https://github.com/gravitational/teleport/actions/runs/10253791543/job/28367216977

#### Relevant snippet

```
=== FAIL: lib/auth TestListUnifiedResources_WithLogins (0.66s)
    auth_with_roles_test.go:4820: 
        	Error Trace:	/__w/teleport/teleport/lib/auth/auth_with_roles_test.go:4820
        	Error:      	Should be empty, but was   []string(Inverse(cmpopts.SortSlices, []string{
        	            	+ 	"fish",
        	            	+ 	"llama",
        	            	  }))
        	Test:       	TestListUnifiedResources_WithLogins
```

### PR DESCRIPTION
**Pull request by gabrielcorado**
Closes #45086

#44611 Changed the app resources to return the `Logins` field (from enriched resources) for AWS Console apps. However, the change needed to be completed, and the field needed to be added in some places.

TLDR: Tests weren't covering this scenario, so this PR also updates them.

Two tests cover those scenarios. However, none of them caught this issue:
- `TestListResources_WithLogins`: `AppServer` wasn't added to the list of resources fetched by the test.
- `TestListUnifiedResources_WithLogins`: This listing process was hard coded to fetch only a limited number of resources, meaning the resulting list would usually never return the apps. (The flakiness on this test was a case where the apps were included on the final list and returned). To improve it, we now consume all resources and assert the final number of resources, so if we add more resources, the test will fail. Hard coding the number of resources could be better. Still, a more scalable solution would require refactoring other tests, so I kept the changes scoped only for the affected tests.

### CODE REVIEW

File: lib/auth/auth_with_roles.go [hunk 1]
```diff
@@ -1777,7 +1777,12 @@ func (a *ServerWithRoles) ListResources(ctx context.Context, req proto.ListResou
 			return trace.Wrap(err)
 		case match:
 			if req.IncludeLogins {
-				logins, err := resourceChecker.GetAllowedLoginsForResource(resource)
+				var checkableResource services.AccessCheckable = resource
+				if appServer, ok := resource.(types.AppServer); ok {
+					checkableResource = appServer.GetApp()
+				}
+
```
Comment (greedy52) [inline lib/auth/auth_with_roles.go:1784]: should this be part of `GetAllowedLoginsForResource`?

File: lib/auth/auth_with_roles.go [hunk 2]
```diff
@@ -2028,21 +2033,26 @@ func (a *ServerWithRoles) listResourcesWithSort(ctx context.Context, req proto.L
 		SearchKeywords: req.SearchKeywords,
 		StartKey:       req.StartKey,
 		EnrichResourceFn: func(r types.ResourceWithLabels) (types.ResourceWithLabels, error) {
-			if req.IncludeLogins && (r.GetKind() == types.KindNode || r.GetKind() == types.KindWindowsDesktop) {
-				resourceChecker, err := a.newResourceAccessChecker(req.ResourceType)
-				if err != nil {
-					return nil, trace.Wrap(err)
-				}
+			if !req.IncludeLogins && (r.GetKind() != types.KindNode || r.GetKind() != types.KindWindowsDesktop || r.GetKind() != types.KindAppServer) {
```
Comment (greedy52) [inline lib/auth/auth_with_roles.go:2036]: nit: not sure how i feel about this check. Do we have to keep expanding this when new types is added to `GetAllowedLoginsForResource`? Could we maybe always call `GetAllowedLoginsForResource` for `req.IncludeLogins` and skip certain errors like `trace.NotImplemented`

**Review by greedy52 — COMMENTED** [review]
Is my understanding correct if this is fixed, then #44983 won't need to use the unified resource api?

**Comment by gabrielcorado** [comment]
> Is my understanding correct if this is fixed, then https://github.com/gravitational/teleport/pull/44983 won't need to use the unified resource api?

That is correct. But, it must use an endpoint that returns enriched resources (`ListResources` or `ListUnifiedResources`).

**Comment by gabrielcorado** [comment]
> I thought the current tc.ListApps calls ListResources

It does, but it does through the [`GetAllResources`](https://github.com/gravitational/teleport/blob/33a54f8e513b7d9aa07aad950f1b76c4c7e81e47/lib/client/api.go#L2529) function which relies on [`GetResourcesPage`](https://github.com/gravitational/teleport/blob/33a54f8e513b7d9aa07aad950f1b76c4c7e81e47/api/client/client.go#L3789) and only returns the resource, not the logins. To get the logins, it should use the [`GetEnrichedResourcesPage`](https://github.com/gravitational/teleport/blob/33a54f8e513b7d9aa07aad950f1b76c4c7e81e47/api/client/client.go#L3721) which uses the same `GetResourcesPage` but maps to `types.EnrichedResource`.

**Comment by greedy52** [comment]
> GetEnrichedResourcesPage

Seems to me a good opportunity to switch `tc` to use that. Well, let's discuss that on other PR.

**Review by greedy52 — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]
